### PR TITLE
don't assert on non-decoded and print null instead of ""

### LINKED ISSUE #814: Missing output for output of `vkGetPhysicalDeviceImageFormatProperties2` from `vkcube`
The `pImageFormatProperties` argument has no body, and therefore isn't valid JSON:
```
"[28]vkGetPhysicalDeviceImageFormatProperties2": {
    "return": "VK_ERROR_FORMAT_NOT_SUPPORTED",
    "physicalDevice": 2,
    "pImageFormatInfo": {
        "sType": "VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_IMAGE_FORMAT_INFO_2",
        "pNext": {
            "sType": "VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_EXTERNAL_IMAGE_FORMAT_INFO",
            "pNext": null,
            "handleType": "VK_EXTERNAL_MEMORY_HANDLE_TYPE_D3D11_TEXTURE_KMT_BIT"
        },
        "format": "VK_FORMAT_R8G8B8A8_UNORM",
        "type": "VK_IMAGE_TYPE_2D",
        "tiling": "VK_IMAGE_TILING_OPTIMAL",
        "usage": 18,
        "flags": 0
    },
    "[out]pImageFormatProperties":
},
```

### PR DESCRIPTION
**Pull request by bradgrantham-lunarg**
Fix #814

### CODE REVIEW

File: framework/generated/vulkan_generators/vulkan_struct_decoders_to_string_body_generator.py [hunk 1]
```diff
@@ -127,10 +127,9 @@ def generate_feature(self):
                 body = inspect.cleandoc('''
                     template <> std::string ToString<decode::Decoded_{0}>(const decode::Decoded_{0}& decoded_obj, ToStringFlags toStringFlags, uint32_t tabCount, uint32_t tabSize)
                     {{
-                        assert(decoded_obj.decoded_value != nullptr);
```
Comment (andrew-lunarg) [inline framework/generated/vulkan_generators/vulkan_struct_decoders_to_string_body_generator.py:130]: Don't we want this in debug builds? Release users will never see this.

**Review by andrew-lunarg — CHANGES_REQUESTED** [review]
There are all the manual versions in `framework/decode/custom_vulkan_struct_decoders_to_string.cpp` too.

**Comment by bradgrantham-lunarg** [comment]
It turns out that null is valid for the decoded value If encoding omitted
the output struct because of a failed result code.  I mentioned that
briefly in the linked issue. So that was asserting on a valid value.

**Comment by bradgrantham-lunarg** [comment]
Good catch! I will go add those.

**Comment by bradgrantham-lunarg** [comment]
> It turns out that null is valid for the decoded value If encoding omitted the output struct because of a failed result code. I mentioned that briefly in the linked issue. So that was asserting on a valid value.

🤔 Do we know in the codegen when a struct is only ever used as an output struct?  I assume we know when a parameter is an output in function codegen, so could we assert `decoded_value

File: framework/generated/vulkan_generators/vulkan_struct_decoders_to_string_body_generator.py [hunk 2]
```diff
@@ -127,10 +127,9 @@ def generate_feature(self):
...
                         if(decoded_obj.decoded_value == nullptr)
                         {{
-                            return "";
+                            return "null";
                         }}
```
[the remark this answers is not in the document]
Comment (bradgrantham-lunarg) [reply framework/generated/vulkan_generators/vulkan_struct_decoders_to_string_body_generator.py:133]: I don't think so, since this is a valid value.

**Pull request merged**